content/docs/guides/testing/_index.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Testing
+meta_desc: "Guide to automated testing of Pulumi programs: unit testing, integration testing, Policy as Code."
+menu:
+    userguides:
+        identifier: testing
+        weight: 7
+---
+
+Pulumi uses general-purpose programming languages to define cloud infrastructure. This capability opens up all the doors to **automated testing** of the infrastructure resources.
+
+The software industry has come up with multiple styles of automated testing. Unit testing of individual pieces, integration testing of system components, end-to-end testing of the complete application&mdash;each style has its advantages and challenges.
+
+Pulumi provides a similar choice of testing options for cloud programs. Currently, we identify three types of tests:
+
+- **Unit Tests** are fast in-memory tests that mock all external calls.
+- **Tests based on Policy as Code** run resource-level assertions *while* infrastructure is being deployed.
+- **Integration Tests** deploy ephemeral infrastructure and run external tests against it.
+
+The following table summarizes the differences between the three approaches:
+
+|                                | Unit Tests  | Policy as Code  | Integration Tests  |
+|--------------------------------|-----|------|------|
+| Provision real infrastructure  | No  | Yes  | Yes  |
+| Require the Pulumi CLI         | No  | Yes  | Yes  |
+| Time to execute                | Milliseconds  | Seconds to minutes  | Seconds to minutes  |
```

Comment:
Perhaps simplify this to "Milliseconds", "Seconds", "Minutes"?  

I guess in part it depends what you think "Policy-as-code" based tests "cost" in terms of time - they are sort of "free" in that they run in amongst your deployment, but you can't get results for them without actually doing a deployment.

Nevertheless, in normal usage, these tests run quickly on the back of incremental changes to infrastructure, whereas integration tests stand up full copies of infrastructure from scratch and tear it down - so being clear these are more heavy-weight feels appropriate.

Reply:
Changed to "seconds"